tests/components/derivative/test_sensor.py

```diff
@@ -354,6 +355,34 @@ async def test_suffix(hass: HomeAssistant) -> None:
     assert round(float(state.state), config["sensor"]["round"]) == 0.0
 
 
+async def test_total_increasing_reset(hass: HomeAssistant) -> None:
+    """Test derivative sensor state with total_increasing sensor input where it should ignore the reset value."""
+    times = [20, 30, 40]
+    values = [10, 30, 0]
```

Comment:
Let's extend with another sample to test the behavior of the state update after a discarded sample